File: Source/NodeJS/Services/Service1/models/DataModels.js
const mongoose = require("mongoose");

const QAModelSchema = new mongoose.Schema(
    {
        projectId: {
            type: String,
            index: true,
            required: true,
            unique: true,
        },
        data: {
            type: String,
            required: true,
            minlength: 10,
            maxlength: 1000
        }
    },
);

const IntentProjectDataSchema = new mongoose.Schema(
    {
        projectId: {
            type: String,
            index: true,
            required: true,
            unique: true,
        },
        data: [
            {
                intent: {
                    type: String,
                    required: true,
                },
                sentences: [String]
            }
        ]
    },
);

const QAModel = mongoose.model("QAModel", QAModelSchema);
const IntentProjectData = mongoose.model("IntentProjectData", IntentProjectDataSchema);

module.exports = {
    QAModel, IntentProjectData
}

File: Source/DotNet/Services/Service2/Models/DataModels.cs
using System.Collections.Generic;

namespace Models
{
    public class QAModel
    {
        public string projectId { get; set; }

        public string data { get; set; }
    }
    public class IntentProjectData
    {
        public string projectId { get; set; }
        public List<IntentData> data { get; set; }
    }
    public class IntentData 
    {
        public string intent { get; set; }
        public List<string> sentences { get; set; }
    }
}